_posts/2017-10-12-Rust-1.21.md

```diff
@@ -0,0 +1,210 @@
+
+```text
+error[E0597]: borrowed value does not live long enough
+  --> src/main.rs:4:14
+   |
+4  |     let x = &5;
+   |              ^ does not live long enough
+...
+10 | }
+   | - temporary value only lives until here
+   |
+   = note: borrowed value must be valid for the static lifetime...
+```
+
+Because the `5` is local, so is its borrow, which doesn't satisfy the
+requirements for `spawn`.
+
+However, if you compile this on Rust 1.21, it will work. Why? Well,
+if the thing being referred to is okay to put into a `static`, we could
+instead de-sugar `let x = &5;` like this:
+
+```rust
+static FIVE: i32 = 5;
+
+let x = &FIVE;
+```
+
+Here, since the `FIVE` is `static`, `x` is a `&'static i32`. And so this
+is what Rust will now do in this kind of case. For full details, see [RFC 1414],
+which was accepted in January, but started in December of 2015!
+
+[RFC 1414]: https://github.com/rust-lang/rfcs/blob/master/text/1414-rvalue_static_promotion.md
+
+We [now run LLVM in parallel while generating
+code](https://github.com/rust-lang/rust/pull/43506), which should reduce peak
+memory usage.
+
+The [RLS](https://github.com/rust-lang-nursery/rls/) can now be installed
+[through rustup](https://github.com/rust-lang/rust/pull/44204) by invoking
+`rustup component add rls-preview`. In general, many useful Rust developer
+tools such as the RLS, Clippy, and `rustfmt` need nightly Rust; this is the
+first steps toward having them work on stable Rust. Please check out the
+preview, and you'll hear more about these plans in the future.
+
+Finally, a few documentation improvements. First up, if you visit [the docs
+for `std::os`] (https://doc.rust-lang.org/stable/std/os/), which contains
+operating-system specific functionality, you'll now see more than just `linux`,
+the platform we build the documentation on. We've long regretted that the hosted
+version of the documentation has been Linux-specific; this is a first step towards
+rectifying that. This is [specific to the standard
+library](https://github.com/rust-lang/rust/pull/43348) and not for general use;
+we'll hope to improve this further in the future.
+
+Next, [Cargo's docs are moving!](https://github.com/rust-lang/rust/pull/43916)
+Historically, Cargo's docs were hosted on doc.crates.io, which doesn't follow
+the release train model, even though Cargo itself does. This lead to situations
+where a feature would land in Cargo nightly, the docs would be updated, and
+then for up to twelve weeks, users would *think* that it should work, but it
+wouldn't yet. [https://doc.rust-lang.org/cargo](https://doc.rust-lang.org/cargo)
+will be the new home of Cargo's docs, though for now, that URL is a redirect to
+doc.crates.io. Future releases will move Cargo's docs over, and at that point,
+doc.crates.io will redirect to doc.rust-lang.org/cargo. Cargo's docs have long
+needed a refreshing, so expect to hear more news about Cargo's docs generally
+through the end of the year.
```

Comment:
Maybe remove "through the end of the year"? I'd expect we'd continue to work on them indefinitely.